### DISCUSSION TITLE
Old surface pro 1

### DISCUSSION THREAD

**Discussion by thomasbjorndahl** [linux-surface/linux-surface#1268] [General]
I know the surface pro version 1 is not on the list, but what’s the reason it’s OFF the list?

**Comment by qzed** [comment]
Mostly because I don't know anything about the device and I haven't seen any reports yet. So I have no idea what works and what doesn't. If you have one on hand, would you mind running our diagnostics script and upload its output (as well as maybe provide some info on what works or what problems it has)? See https://github.com/linux-surface/linux-surface/wiki/Testers-and-Contributing#support-for-new-devices for details on that (note that the diagnostics script also collects an acpidump, so you don't have to collect it separately).

**Reply by thomasbjorndahl** [reply]
I would love to run the script, but I'm not smart enough to know how to run them.
I have downloaded the latest git  (for windows) but don't know where to go from there.

**Reply by qzed** [reply]
Ah, the script won't work on windows. You'd have to try and install Linux first. If that works successfully, you can download the script from [here](https://raw.githubusercontent.com/linux-surface/linux-surface/master/scripts/diagnostics.sh) and run it via `./diagnostics.sh full` in the terminal. It should create an archive that you can then upload here.

**Reply by thomasbjorndahl** [reply]
I have sucessfully installed ubuntu and have run the script.
But it stops

thomas@surface-UBUNTU:~/Downloads$ sudo ./diagnostics.sh full
 ==> collecting information...
   - uname
   - DMI system information
   - dmesg log
   - ACPI dump
sudo: acpidump: command not found
   - ACPI device information and paths (sysfs)
./diagnostics.sh: line 125: tree: command not found

**Reply by sphh** [reply]
On Ubuntu, you should be able to install the `tree` package. This can be done e.g. with `sudo apt install tree` from the command line or with the graphical package manager.

**Reply by qzed** [reply]
`acpidump` seems to be missing as well. That should be available via the `acpica-tools` package.

**Reply by qzed** [reply]
I should probably improve the script and add checks for the packages...

**Reply by thomasbjorndahl** [reply]
Great. That worked. Where can I upload ?

**Reply by thomasbjorndahl** [reply]
Here?
[diagnostics.tar.gz](https://github.com/linux-surface/linux-surface/files/13034462/diagnostics.tar.gz)

**Reply by qzed** [reply]
Thanks! Any specific things that don't work?

**Reply by thomasbjorndahl** [reply]
Most works. Even the touch pad on the extra keyboard.
But I haven't seen the front camera working. Only the back.

**Reply by qzed** [reply]
It looks like there are
```
Bus 001 Device 005: ID 045e:0794 Microsoft Corp. Front LifeCam
Bus 002 Device 003: ID 045e:0795 Microsoft Corp. Rear LifeCam
```
which both seem to be UVC compatible cameras. I can't see any related errors in the log, so I'm not sure why one is working but the other isn't.

**Reply by qzed** [reply]
I've added the Pro 1 to the readme.

**Reply by thomasbjorndahl** [reply]
After a re boot I got both cameras :-)
All seams to work here...
Thank's a lot
